### ISSUE TITLE
Error when using customed clusters = '...'

### ISSUE THREAD

**Issue by zehualilab** [plger/scDblFinder#7]
Thanks for this tool! I met this problem when I tried to set customed clusters = '...' during running :
scDblFinder(sce, samples = "batch", clusters = 'my_cluster_label')
Error in scDblFinder(sce[, x], artificialDoublets = artificialDoublets,  : 
  Only one cluster generated

my_cluster_label is a colname in the colData(sce)
I tried to convert the class or type of sce@colData$my_cluster_labelto factors/numeric/characters or set clusters = sce@colData$my_cluster_label but they all turn out to be useless.
I could get the 'scDblFinder.class' label in my sce without setting clusters.
I was using scDblFinder V1.1.8.
Thanks for any help!!!

**Comment by plger** [comment]
Hi,
thanks for reporting this, I thought I had solved that bug before though... you say version 1.1.8, but current version is 1.1.6 - can you double check that?
Pierre-Luc

**Comment by zehualilab** [comment]
Thanks for the quick reply. Here is my package description from R 3.6.3 and bioconductor 3.10:
Package: scDblFinder
Type: Package
Title: scDblFinder
Version: 1.1.8
Depends: R (>= 3.6)
Authors@R: person("Pierre-Luc", "Germain", email="pierre-luc.germain@hest.ethz.ch", role=c("cre","aut"), comment=c(ORCID="0000-0003-3418-4218"))
URL: https://github.com/plger/scDblFinder
BugReports: https://github.com/plger/scDblFinder/issues
Description: Efficient identification of doublets in single-cell RNAseq directly from counts using overclustering-based generation of artifical doublets.
License: GPL-3
Imports: igraph, Matrix, matrixStats, BiocParallel, BiocNeighbors,
        SummarizedExperiment, SingleCellExperiment, scran, scater,
        data.table, dplyr, ggplot2, randomForest, graphics, methods,
        stats, DelayedArray
Suggests: BiocStyle, knitr, rmarkdown
VignetteBuilder: knitr
Encoding: UTF-8
RoxygenNote: 6.1.1
biocViews: Preprocessing, SingleCell, RNASeq
git_url: https://git.bioconductor.org/packages/scDblFinder
git_branch: RELEASE_3_10
git_last_commit: f9806b7
git_last_commit_date: 2020-03-05
Date/Publication: 2020-03-05
NeedsCompilation: no
Packaged: 2020-03-06 06:51:06 UTC; biocbuild
Author: Pierre-Luc Germain [cre, aut] (<https://orcid.org/0000-0003-3418-4218>)
Maintainer: Pierre-Luc Germain <pierre-luc.germain@hest.ethz.ch>
Built: R 3.6.3; ; 2020-03-06 13:43:44 UTC; windows

I tried to re-install after receiving your reply. When I tried to use R 4.0.0 and bioconductor 3.11 to install the devel version of it, the version was 1.1.6. When I tried to use R 3.6.3 and bioconductor 3.10 to install, the version was still 1.1.8. Devtools installation of scDblFinder returns V 1.1.6 for now. 

I tired running this step again in V 1.1.6 and it worked well. V 1.1.8 still caused this error. I remember I gave up the 1.1.6 version and choose the 1.1.8 version before.
